rust/generated_parser/src/ast_builder.rs

```diff
@@ -133,7 +134,7 @@ impl<'alloc> AstBuilder<'alloc> {
         token: arena::Box<'alloc, Token<'alloc>>,
     ) -> Result<'alloc, arena::Box<'alloc, Identifier<'alloc>>> {
         let loc = token.loc;
-        self.on_identifier_reference(token.value.unwrap(), loc.start)?;
+        self.on_identifier_reference(token.terminal_id, token.value.unwrap(), loc.start)?;
```

Comment:
I think we can always pass the token by reference, even in the "yield" and "await" cases.